### PR TITLE
Adds analytics and user conf

### LINKED ISSUE #25: Squash danger vibes
The use of red for the log responses triggers error responses in developers (me). I'd suggest using blue or magenta maybe?

### PR DESCRIPTION
**Pull request by abea**
Adds some basic usage tracking, with support of added user configuration.

Also closes #25 and #35 

- [x] Add a `postuninstall` script to remove the config file.

### CODE REVIEW

**Comment by abea** [comment]
@boutell If you're still able, would you please test this on your PC?

**Comment by abea** [comment]
Any red flags yet?

**Comment by abea** [comment]
We're not merging into master just yet, to not load this one PR up with too much that should be deployed in that release.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
 
-## 2.3.4 - May 18, 2020
+## 2.4.0 - 2020-06-03
 
-* Documents and prioritizes the existing `apos` command alias as the CLI command.
+* Documents and prioritizes the existing `apos` command alias as the CLI
+command.
+* Adds the opt-in option to send basic, anonymous usage information to the
+maintainers. This involves the CLI creating a configuration file on the user's
+machine to save the user's approval response and a unique ID.
```
Comment (boutell) [inline CHANGELOG.md:9]: "a random unique ID."

File: README.md [hunk 2]
```diff
@@ -71,15 +71,19 @@ apos create-module <module name>
 
 Remember to register the module in `apps.js` with the other module types.
 
+## CLI analytics
+
+To help us better understand how devs are using the CLI, you will have the option to *opt-in* to share basic usage information with the maintainers. This is limited to 1) the command you have executed, 2) if that command succeeded, and 3) a random, anonymous user ID.
```
Comment (boutell) [inline README.md:76]: Whoah, that sounds like they can tell when people run apostrophe command line tasks like creating users and groups. I think that impression is scary to give (and we need to make sure it's not true, too).

Honestly I think we should dial back our ambition here and collect solely when a site is created, so we can say "that's all we gather, and we do NOT record the name of the site, so don't worry."

File: README.md [hunk 3]
```diff
@@ -71,15 +71,19 @@ apos create-module <module name>
...
+
+**If you do not explicitly agree to participate, no infomation will be tracked.** Through the `universal-analytics` package, we use the option to anonymize your IP address, so we know as little as possible about you other than what is listed above. See the `sendAnalyticsEvent` utility method if you're curious how this all works.
```
Comment (boutell) [inline README.md:78]: This is good info.

File: bin/apostrophe [hunk 4]
```diff
@@ -2,60 +2,67 @@
...
-  return output;
-}
+  util.checkDependencies();
 
-util.checkDependencies();
+  require('../lib/commands/create')(program);
+  require('../lib/commands/create-widget')(program);
+  require('../lib/commands/create-piece')(program);
+  require('../lib/commands/create-module')(program);
+  var aposCmd = require('../lib/apostrophe')(program);
```
Comment (boutell) [inline bin/apostrophe:51]: Apostrophe 2 no longer runs on Node 6, so you may as well go const/let, but don't let it slow you down, this is great.

**Comment by boutell** [comment]
I wanted to say, since I raised concerns and it might not be otherwise clear: this is very good work.

File: README.md (hunk 2 above)
Comment (abea) [reply README.md:76]: I can definitely be more clear. We are intending to get some data on `create-piece`, `create-widget`, and `create-module` as well. But only the CLI-specific stuff. Honestly didn't we come the conclusion that the CLI wasn't super useful for other apostrophe tasks?

File: bin/apostrophe (hunk 4 above)
Comment (abea) [reply bin/apostrophe:51]: Totally. I didn't want to clutter this PR up with all those diffs. Certainly a quick cleanup project for the future.

**Comment by abea** [comment]
@localghost443 See https://github.com/apostrophecms/apostrophe-cli/pull/32/commits/8ff56fb8f9d537b7a2553fe18d06894734b2a423.

Are there any blockers to getting this merged for the next release?

**Comment by abea** [comment]
Another example of CLI usage tracking: https://github.com/nuxt/telemetry

**Pull request closed without merging**